### ISSUE TITLE
Detected that custom integration 'momentary' calls async_write_ha_state from a thread at custom_components/momentary/switch.py...

### ISSUE THREAD

**Issue by glc650** [twrecked/hass-momentary#37]
Since updating to HA Core 2024.5.0 I now this error in the HA logs:

```
Logger: homeassistant.components.websocket_api.http.connection
Source: components/websocket_api/commands.py:239
integration: Home Assistant WebSocket API (documentation, issues)
First occurred: 5:26:48 AM (1 occurrences)
Last logged: 5:26:48 AM

[281472740648384] Unexpected exception
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 239, in handle_call_service
    response = await hass.services.async_call(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2738, in async_call
    response_data = await coro
                    ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2779, in _execute_service
    return await target(service_call)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 975, in entity_service_call
    single_response = await _handle_entity_call(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 1047, in _handle_entity_call
    result = await task
             ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1680, in async_turn_on
    await self.hass.async_add_executor_job(ft.partial(self.turn_on, **kwargs))
  File "/usr/local/lib/python3.12/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/momentary/switch.py", line 213, in turn_on
    self._start_activity(True)
  File "/config/custom_components/momentary/switch.py", line 210, in _start_activity
    self.async_schedule_update_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1258, in async_schedule_update_ha_state
    self.async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1008, in async_write_ha_state
    self.hass.verify_event_loop_thread("async_write_ha_state")
  File "/usr/src/homeassistant/homeassistant/core.py", line 440, in verify_event_loop_thread
    frame.report(
  File "/usr/src/homeassistant/homeassistant/helpers/frame.py", line 162, in report
    _report_integration(what, integration_frame, level, error_if_integration)
  File "/usr/src/homeassistant/homeassistant/helpers/frame.py", line 203, in _report_integration
    raise RuntimeError(
RuntimeError: Detected that custom integration 'momentary' calls async_write_ha_state from a thread at custom_components/momentary/switch.py, line 210: self.async_schedule_update_ha_state(). Please report it to the author of the 'momentary' custom integration.
```

```
Logger: homeassistant.helpers.frame
Source: helpers/frame.py:190
First occurred: 5:26:48 AM (1 occurrences)
Last logged: 5:26:48 AM

Detected that custom integration 'momentary' calls async_write_ha_state from a thread at custom_components/momentary/switch.py, line 210: self.async_schedule_update_ha_state(), please report it to the author of the 'momentary' custom integration
```

whenever one of my momentary switches is turned on.  I was running version 0.6.3 but just updated to 0.7.0.alpha.6 and still have the issue.

**Comment by twrecked** [comment]
I had the same issue `hass-aarlo`; I'll push a fix.

**Comment by twrecked** [comment]
Can you try the latest 0.7 build? It should be fixed by #38

**Comment by glc650** [comment]
.7 is working, thanks

**Comment by lzzsalvo** [comment]
the latest release (7b8) works fine. Thank you

**Comment by twrecked** [comment]
Should be fixed in both 0.7 and 0.6
